### PR TITLE
predict_joint

### PR DESCRIPTION
**Pull request by cscherrer**
Not tested yet, but I think this is right.

Compare:
```julia
function MMI.predict(sm::SossMLJModel, fitresult, Xnew)
    m = sm.model
    post = fitresult.post
    pred = predictive(m, keys(post[1])...)
    
    map(Tables.rowtable(Xnew)) do xrow
        args = merge(sm.transform([xrow]), sm.hyperparams)
        SossMLJPredictor(sm, post, pred, args)
    end
end

function MMI.predict_joint(sm::SossMLJModel, fitresult, Xnew)
    m = sm.model
    post = fitresult.post
    pred = predictive(m, keys(post[1])...)
    args = merge(sm.transform(Xnew), sm.hyperparams)
    return SossPredictor(sm, post, pred, args)
end
```

### CODE REVIEW

File: src/SossMLJ.jl [hunk 1]
```diff
@@ -97,12 +97,18 @@ function MMI.predict(sm::SossMLJModel, fitresult, Xnew)
...
 
-    # predictor = SossPredictor(sm, post, pred, Xnew)
-
     map(Tables.rowtable(Xnew)) do xrow
         args = merge(sm.transform([xrow]), sm.hyperparams)
         SossMLJPredictor(sm, post, pred, args)
     end
 end
 
+function MMI.predict_joint(sm::SossMLJModel, fitresult, Xnew)
```
Comment (DilumAluthge) [inline src/SossMLJ.jl]:
```suggestion
# function MMI.predict_joint(sm::SossMLJModel, fitresult, Xnew)
function predict_joint(sm::SossMLJModel, fitresult, Xnew)
```

`MMI.predict_joint` doesn't actually exist yet upstream. So let's leave this comment as a reminder to ourself.

Comment (DilumAluthge) [reply src/SossMLJ.jl]: @cscherrer

**Comment by DilumAluthge** [comment]
@cscherrer Could you explain the difference between `SossPredictor` and `SossMLJPredictor`?

**Comment by DilumAluthge** [comment]
Nightly failures are not related to this PR.

**Comment by cscherrer** [comment]
> @cscherrer Could you explain the difference between `SossPredictor` and `SossMLJPredictor`?

Oops typo, the latter is correct

**Review by DilumAluthge — APPROVED** [review]
LGTM

**Pull request merged**